### PR TITLE
chore: refactor exception strings into enums

### LINKED ISSUE #1793: Update thrown errors to use enums
As part of cleanup thrown errors should be refactored to utilize enums or constants instead of magic strings.

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1793  🦕

### CODE REVIEW

**Review by shaffeeullah — APPROVED** [review]
Amazing, thanks Denis!

**Review by danielbankhead — APPROVED** [review]

**Comment by danielduhh** [comment]
Please add an issue for this! @ddelgrosso1

File: src/bucket.ts [hunk 1]
```diff
@@ -1401,11 +1414,11 @@ class Bucket extends ServiceObject {
     callback?: CombineCallback
   ): Promise<CombineResponse> | void {
     if (!Array.isArray(sources) || sources.length === 0) {
-      throw new Error('You must provide at least one source file.');
+      throw new Error(BucketExceptionMessages.PROVIDE_SOURCE_FILE);
```
Comment (danielduhh) [inline src/bucket.ts:1417]: Unrelated to the refactor: this looks like client side validation and it concerns me... Are these errors that we should be deferring to the server?

**Comment by ddelgrosso1** [comment]
@danielduhh added issue #1793 that will be closed by this PR.

**Pull request merged**